client.go

```diff
@@ -5,13 +5,23 @@ import "github.com/kochavalabs/mazzaroth-xdr/xdr"
 // Client defines a mazzaroth client that exposes common
 // endpoints exposed by a mazzaroth readonly node.
 type Client interface {
-	TransactionSubmit(transaction xdr.Transaction) (*xdr.TransactionSubmitResponse, error)
-	TransactionLookup(transactionID xdr.ID) (*xdr.TransactionLookupResponse, error)
-	ReceiptLookup(transactionID xdr.ID) (*xdr.ReceiptLookupResponse, error)
-	BlockLookup(blockID xdr.Identifier) (*xdr.BlockLookupResponse, error)
-	BlockHeaderLookup(blockID xdr.Identifier) (*xdr.BlockHeaderLookupResponse, error)
-	AccountInfoLookup(accountID xdr.ID) (*xdr.AccountInfoLookupResponse, error)
-	ChannelInfoLookup(channelInfoType xdr.ChannelInfoType) (*xdr.ChannelInfoLookupResponse, error)
-	BlockHeightLookup() (uint64, error)
-	AbiLookup(channelID xdr.ID) (*xdr.Abi, error)
+	TransactionSubmitCall(channelID string, seed string, functionName string, parameters []string, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionSubmitContract(channelID string, seed string, contractBytes []byte, abiDef []byte, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionSubmitConfig(channelID string, seed string, owner string, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionUpdatePermission(channelID string, seed string, nonce uint64, blockExpirationNumber uint64, authorizedAddressStr string, alias string, authorizedAlias string, authorize bool) (*xdr.Response, error)
```

Comment:
Don't you find this model less confusing than the one parameter fits all one? Looking at the interface it can clearly be seen that there are four possible operations and the parameters each one takes in. No need to create a builder object, just pass in the required values. Internally they are using the builder.

Please, confirm you still prefer to have just one parameter with the builder.

Reply:
I'm going to revert the change in another branch, but before integrating it in here I'd like to discuss this as I see this way much simpler.

Reply:
```jmi/upgrade-rest-builder``` this branch is the version with builder. We can merge it into this one if you find that model better.

Reply:
Personally, I like the single function version with the users using the Transaction Builder.  Might be a question for @elewis787

Reply:
The single parameter allows for two options - building the tx object manually then passing it to the function ( I see this as equal to passing multiple parameters ) or using the builder to build a tx object directly. 

With these two options I would like to see the interface remain small with a transaction function that takes a transaction object.

Reply:
I've hard reset this branch up to the single parameter one. The code is ready for review again. Thanks.